src/common/utilities.ts

```diff
@@ -73,33 +73,63 @@ export async function getProjectRoot(): Promise<WorkspaceFolder> {
     }
 }
 
-export async function installDependenciesIfNeeded(context: vscode.ExtensionContext): Promise<void> {
-    // Install necessary dependencies for the flowcharts and telemetry
+export async function installTelemetryDependenciesIfNeeded(context: vscode.ExtensionContext): Promise<void> {
+    // Install necessary dependencies for the telemetry
     const alreadyInstalled = context.globalState.get(DEPENDENCIES_INSTALLED, false);
 
     if (!alreadyInstalled) {
-        const vizPathToScript = 'bundled/tool/install_viz_dependencies.py';
         const telemetryPathToScript = 'bundled/tool/install_telemetry_dependencies.py';
         try {
-            const stdoutViz = await callPythonScript(vizPathToScript, EXTENSION_ROOT_DIR, context);
             const stdoutTelemetry = await callPythonScript(telemetryPathToScript, EXTENSION_ROOT_DIR, context);
             // Check if the script output contains the success message
-            if (stdoutViz.includes('Successfully installed')) {
-                traceLog(`Kedro-viz dependencies installed!`);
-                console.log('Kedro-viz dependencies installed!');
-            }
             if (stdoutTelemetry.includes('Successfully installed')) {
-                traceLog(`kedro-telemetry dependencies installed!`);
-                console.log('kedro-telemetry dependencies installed!');
+                traceLog(`kedro-telemetry and its dependencies installed!`);
+                console.log('kedro-telemetry  and its dependencies installed!');
             }
             context.globalState.update(DEPENDENCIES_INSTALLED, true);
         } catch (error) {
-            traceError(`Failed to install Python dependencies:: ${error}`);
-            console.error(`Failed to install Python dependencies:: ${error}`);
+            traceError(`Failed to install kedro-telemetry and its dependencies:: ${error}`);
+            console.error(`Failed to install kedro-telemetry and its dependencies:: ${error}`);
         }
     }
 }
 
+export async function installKedroViz(context: vscode.ExtensionContext): Promise<boolean> {
+    const vizPathToScript = 'bundled/tool/install_viz_dependencies.py';
+    try {
+        const stdoutTelemetry = await callPythonScript(vizPathToScript, EXTENSION_ROOT_DIR, context);
+
+        // Check if the script output contains the success message
+        if (stdoutTelemetry.includes('Successfully installed')) {
+            traceLog('Kedro-Viz and its dependencies installed!');
+            console.log('Kedro-Viz and its dependencies installed!');
+            return true;
+        }
+        context.globalState.update(DEPENDENCIES_INSTALLED, true);
```

Comment:
Is this sharing the same global state with telemetry dependencies? I see it update the state but not consuming it here.

Reply:
@noklam sorry I forgot to remove this line `context.globalState.update(DEPENDENCIES_INSTALLED, true);` I will update it .